container-engine-lib/lib/backend_interface/objects/service/service.go

```diff
@@ -8,12 +8,20 @@ import (
 )
 
 const (
-	ServiceNameRegex            = "[a-zA-Z0-9-_]+"
-	wordWrappedServiceNameRegex = "^" + ServiceNameRegex + "$"
+	// ServiceNameRegex implements RFC-1123 for naming services, namely:
+	// * contain at most 63 characters
+	// * contain only lowercase alphanumeric characters or '-'
+	// * start with an alphanumeric character
+	// * end with an alphanumeric character
+	// The adoption of RFC-1123 is to maintain compatability with current Kubernetes service and pod naming standards:
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+	ServiceNameRegex            = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
```

Comment:
I _think_ the following should work:
```
^[a-z0-9][-a-z0-9]{0,61}[a-z0-9]$
```
and it has the advantage of encoding the length constraint inside the regexp itself, so that you don't need to have this separate function `isServiceNameCorrectLength` below. This would be a bit cleaner IMO

Reply:
@gbouv I struggled with this myself and thought adding the length in the regex was better as well. _But_ the challenge is that with this regex you are requiring a minimum of 2 characters, while the spec says that the name length could be a single character. I thought about just making it so that _we_ would have a minimum of 2 chars, but didn't want to diverge from the kubernetes standard. I felt that it was simpler to just check the length. Let me know if you see it differently but I weighed alignment with kubernetes and RFC 1123 as more important.

Reply:
Perfect, I missed the `?` at the end as an option. Thanks for suggesting this!